docker_cli/image.py

```diff
@@ -0,0 +1,458 @@
+            context_str,
+            "-t",
+            tag
+        ]
+
+        if no_cache:
+            cmd += ["--no-cache"]
+
+        if dockerfile_build:
+            # If a dockerfile path is given, include it.
+            # Else, docker build will default to "./Dockerfile"
+            if dockerfile is not None:
+                cmd += [f"--file={os.fspath(dockerfile)}"]
+            stdin = None
+        else:
+            cmd += ["-f-"]
+            stdin = dockerfile_string
+
+        run(
+            cmd,
+            text=True,
+            stdout=stdout,  # type: ignore
+            stderr=stderr,  # type: ignore
+            input=stdin
+        )
+
+        return cls(tag)
+
+    def _inspect(self, format: Optional[str] = None) -> str:
+        """
+        Use 'docker inspect' to retrieve a piece of information about the
+        image.
+
+        Parameters
+        ----------
+        format : str, optional
+            The value to be requested by the --format argument, or None.
+            Defaults to None.
+
+        Returns
+        -------
+        str
+            The string returned by the docker inspect command.
+
+        Raises
+        -------
+        CalledProcessError
+            If the docker inspect command fails.
+        """
+        cmd = ["docker", "inspect"]
+        if format:
+            cmd += [f"-f={format}"]
+        cmd += [self._id]
+
+        inspect_result = run(
+            cmd,
+            capture_output=True,
+            text=True,
+            check=True
+        )
+
+        output_text = inspect_result.stdout
+        return output_text
+
+    def run(
+        self,
+        command: str,
+        *,
+        stdout: Optional[Union[io.TextIOBase, int]] = None,
+        stderr: Optional[Union[io.TextIOBase, int]] = None,
+        interactive: bool = False,
+        network: str = "host"
+    ):
+        """
+        Run the given command on a container.
+
+            .. note::
+        For stderr special values to pass into `stdout` and `stdin`, review
+        values passable into the same arguments in the :func:`~subprocess.run`
+        function:
+        https://docs.python.org/3/library/subprocess.html#subprocess.run
+
+            .. warning::
+        This method does not work correctly if the image built does not have
+        bash installed.
+
+        Parameters
+        ----------
+        cmd : str
+            The desired command, in linux command format.
+        stdout : io.TextIOBase or subprocess special value or None, optional
+            The location to send the process stdout output to.Defaults to None.
+        stderr : io.TextIOBase or subprocess special value or None, optional
+            The location to send the process stderr output to.
+            Defaults to None.
+        interactive : bool, optional
+            A boolean describing whether or not to run this command in
+            interactive mode or not. Defaults to True.
```

Comment:
Currently, this defaults to False, and I think that's my preferred behavior.

```suggestion
            interactive mode or not. Defaults to False.
```